File: java/src/SortArrayByParity.java
public class SortArrayByParity {
    public int[] sortArrayByParity(int[] array) {
        int[] result = new int[array.length];
        for (int left = 0, right = array.length - 1, index = 0 ; index < array.length ; index++) {
            if (array[index] % 2 == 0) {
                result[left++] = array[index];
            } else {
                result[right--] = array[index];
            }
        }
        return result;
    }
}

File: python/sort_array_by_parity.py
from typing import List


class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        result = [0] * len(nums)
        left, right, index = 0, len(nums) - 1, 0
        while index < len(nums):
            if nums[index] % 2 == 0:
                result[left] = nums[index]
                left += 1
            else:
                result[right] = nums[index]
                right -= 1
            index += 1
        return result
